circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/S3S3Copier.java

```diff
@@ -156,7 +156,7 @@ private void startAllCopyJobs() {
     AmazonS3URI targetBase = toAmazonS3URI(replicaLocation.toUri());
     srcClient = s3ClientFactory.newInstance(sourceBase, s3s3CopierOptions);
     targetClient = s3ClientFactory.newInstance(targetBase, s3s3CopierOptions);
-    transferManager = transferManagerFactory.newInstance(targetClient, s3s3CopierOptions);
+    retryableTransferManager = retryableTransferManagerFactory.newInstance(targetClient, s3s3CopierOptions, srcClient);
```

Comment:
I see that `newInstance` is taking in `srcClient` as the argument now. Do we know why was it not like this in `TransferManagerFactory`. Just want to understand, if there was any specific reason for not tying up the transferManager to a sourceClient. If this coupling is unnecessary, than I think, it should be avoided.

@patduin thoughts ?

Reply:
it was already tied just went from `copy` see line 214 to the`newInstance` method. That's ok it is a field so makes more sense to load it in here. The copy method now has all arguments with the same scope.